### PR TITLE
redefined multiple structs

### LINKED ISSUE #440: Update all code to match the new SQL tables
### Description
Update all frontend and backend code to represent the new SQL tables we designed

### Definition of Done
- Include FoodType string variable
- remove garbage variables from Ingredient type in frontend and FoodItem type in backend
- remove PantryLastUpdated time

### PR DESCRIPTION
**Pull request by rileycleavenger**
## Describe your changes
I redefined all of the Golang structs and some of the TypeScript structs to better align with the new database tables

closes #440

### CODE REVIEW

**Comment by rileycleavenger** [comment]
Here are all of the updated structs in backend 

```
type FoodItem struct {
	Name      string `json:"Name"`
	FoodType string `json:"FoodType"`
	SaleDetails string `json:"SaleDetails"`
	Quantity int `json:"Quantity"`
}
```
```
type GroceryStore struct {
	Name      string
	ZipCode   string
}
```
```
type List struct {
	ShoppingList []FoodItem
}
```
```
type Pantry struct {
	FoodInPantry []FoodItem
}
```
```
type Recipe struct {
    Instructions   string   `json:"instructions"`
    Ingredients    []string `json:"ingredients"`
    Title          string   `json:"title"`
    PictureLink    *string  `json:"pictureLink"`
    RecipeID        string   `json:"recipeID"`
    UserFavorite   bool     `json:"userFavorite"`
}
```
```
type Scraper struct {
	PublixDeals []FoodItem
	WalmartDeals []FoodItem
}
```
```
type User struct{
	FirstName string `json:"FirstName"`
	LastName string `json:"LastName"`
	Email string `json:"Email"`
        UserName string `json:"UserName"`
        Password string `json:"Password"`
	UserPantry Pantry 
	UserList List 
}
```

File: Saucier720-app/src/app/core/interfaces/store.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export interface Store {
     Name: string;
+    ZipCode: string;
```
Comment (TylerMetz) [inline Saucier720-app/src/app/core/interfaces/store.ts:3]: Zipcode here? I think this isn't a camel case thing as it's one word

File: Saucier720-app/src/app/list/list.component.ts [hunk 2]
```diff
@@ -64,11 +64,9 @@ export class ListComponent implements OnInit {
     if (this.newIngredientName && this.newIngredientQuantity > 0) {
       const newIngredient: Ingredient = {
         Name: this.newIngredientName,
-        Quantity: this.newIngredientQuantity,
-        StoreCost: 0, // Example value, replace with actual value if needed
-        OnSale: false, // Example value, replace with actual value if needed
-        SalePrice: 0, // Example value, replace with actual value if needed
-        SaleDetails: '' // Example value, replace with actual value if needed
+        FoodType: '', // Example value, replace with actual value if needed
```
Comment (TylerMetz) [inline Saucier720-app/src/app/list/list.component.ts:67]: Is there a better word than "example value" here, it's not an example as there is nothing there but an empty string

File: src/BackendPkg/User.go [hunk 3]
```diff
@@ -11,17 +11,10 @@ type User struct{
 	Email string `json:"Email"`
     UserName string `json:"UserName"`
```
Comment (TylerMetz) [inline src/BackendPkg/User.go:12]: I guess another thing here and in our tables. Is it UserName or Username with camel case?

File: src/BackendPkg/Ingredient.go
Comment (TylerMetz) [inline src/BackendPkg/Ingredient.go:1]: Rip

**Review by TylerMetz — APPROVED** [review]
LGTM! Left a few annoying comments because you made me do this in my bed on GitHub mobile

File: src/BackendPkg/Ingredient.go
Comment (rileycleavenger) [reply src/BackendPkg/Ingredient.go:1]: rip

File: src/BackendPkg/User.go (hunk 3 above)
Comment (rileycleavenger) [reply src/BackendPkg/User.go:12]: We  did "UserName" in the tables

File: Saucier720-app/src/app/core/interfaces/store.ts (hunk 1 above)
Comment (rileycleavenger) [reply Saucier720-app/src/app/core/interfaces/store.ts:3]: it's two words

File: Saucier720-app/src/app/list/list.component.ts (hunk 2 above)
Comment (rileycleavenger) [reply Saucier720-app/src/app/list/list.component.ts:67]: ok, i'll change and recommit

**Pull request merged**